### PR TITLE
Bug 1244682 - Base Modularize HomescreenLauncher and LandingLauncher

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=64a739ca51c7ac7681cbb8071dc4eabe775bac38

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=278a7083ca3450eaf142ae1c0bac037f79aa6f23

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1b1e2dcd1ad224e1dec5ad4677da2b37ef7e07be

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f44bf2cf66641767ae5526c3da36c155cd2fdf95

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5cfcb7fd4e425e489d7d6d313c5040790f71b82f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=07cf424f29adf529f077a2291dccf66530e8480f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=477788ae7c42908b9c1dc81d69715afa0dd9f772

File: tv_apps/smart-system/js/homescreen_launcher.js [hunk 1]
```diff
@@ -24,6 +25,16 @@
   var HomescreenLauncher = function() {
   };
 
+  HomescreenLauncher.EVENTS = [
```
Comment (luke-chang) [inline tv_apps/smart-system/js/homescreen_launcher.js]: Due to the utilization of BaseModule's `EVENTS` object here, the original `handleEvent` method [1] will be overwritten by BaseModule [2]. It turns out that the events don't seem to be handled by `HomescreenLauncher` after this patch. I guess we need to replace its name with `_pre_handleEvent / _post_handleEvent` or follow the convention of `_handle_<event name>` to address each single event.

[1] https://github.com/albertopq/gaia/blob/1244682-base-module-launchers/tv_apps/smart-system/js/homescreen_launcher.js#L133
[2] https://github.com/mozilla-b2g/gaia/blob/master/tv_apps/smart-system/js/base_module.js#L333

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5fd67564728301270579c1539c8b9ed812e71f66

File: tv_apps/smart-system/js/homescreen_launcher.js (hunk 1 above)
Comment (albertopq) [inline tv_apps/smart-system/js/homescreen_launcher.js]: Despite you are right about handleEvent not being called, your statement is not totally true. handleEvent won't be  overwritten by BaseModule, but the other way around. The problem here is that we are overwriting start/stop methods, and baseModule start method is the one that adds the listeners [1]. Using _start/_stop instead should fix the issue. 

Thanks for the review, that was a good catch!

[1] https://github.com/mozilla-b2g/gaia/blob/master/tv_apps/smart-system/js/base_module.js#L708

Comment (luke-chang) [inline tv_apps/smart-system/js/homescreen_launcher.js]: Yes, you're right! I misunderstood the `BaseModule.mixin()`. I'll look into it soon.

**Pull request merged**